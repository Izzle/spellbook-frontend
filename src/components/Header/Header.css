.Header__not-logged-in {
    display: flex;
    flex: 1;
    align-items: center;
    justify-content: flex-end;
  }