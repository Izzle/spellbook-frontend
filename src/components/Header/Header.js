import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Nav from '../Nav/Nav';
import TokenService from '../../services/token-service';
import './Header.css';

export default class Header extends Component {

  handleLogoutClick = () => {
    TokenService.clearAuthToken();
  }

  renderLoginLink() {
    return (
      <div className='Header__not-logged-in'>
        <Link to='/login' className='Header__Link'>
          Log In
        </Link>
        <Link to='/register' className='Header__Link'>
          Register
        </Link>
      </div> 
    )
  }

  renderLogoutLink() {
    return (
      <div className='Header__logged-in'>
        <Link
          onClick={this.handleLogoutClick}
          to='/'
          className='Header__Link'>
        Logout</Link>
      </div>
    )
  }


  render() {
      return <>
        <nav className='Header'>
          <Nav />
          <h1>
              <Link to='/'>Spellbook</Link>
          </h1>
          {TokenService.hasAuthToken()
            ? this.renderLogoutLink()
            : this.renderLoginLink()}
        </nav>
      </>
  }
}