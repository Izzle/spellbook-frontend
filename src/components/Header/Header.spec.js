import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import Header from './Header';
import { BrowserRouter } from 'react-router-dom';


describe('<Header />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BrowserRouter><Header /></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><Header /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});