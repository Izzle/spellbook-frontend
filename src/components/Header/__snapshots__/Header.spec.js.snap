// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders this UI as expected 1`] = `
<nav
  className="Header"
>
  <ul
    className="Nav__list"
  >
    <li>
      <a
        className="Nav__Link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className="Nav__Link"
        href="/spells"
        onClick={[Function]}
      >
        Spells
      </a>
    </li>
    <li>
      <a
        className="Nav__Link"
        href="/spell-creator"
        onClick={[Function]}
      >
        Create a Spell
      </a>
    </li>
    <li>
      <a
        className="Nav__Link"
        href="/spellbook"
        onClick={[Function]}
      >
        View Spellbooks
      </a>
    </li>
  </ul>
  <h1>
    <a
      href="/"
      onClick={[Function]}
    >
      Spellbook
    </a>
  </h1>
  <div
    className="Header__not-logged-in"
  >
    <a
      className="Header__Link"
      href="/login"
      onClick={[Function]}
    >
      Log In
    </a>
    <a
      className="Header__Link"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</nav>
`;
